tests/rptest/scale_tests/basic_auth_test.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from concurrent.futures import ThreadPoolExecutor
+import json
+import random
+import requests
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from rptest.services.redpanda import SecurityConfig
+from rptest.tests.pandaproxy_test import PandaProxyEndpoints, Consumer
+from typing import Optional, List, Dict, Union
+
+
+class User:
+    def __init__(self, topic: str):
+        self.username = f'user_{TopicSpec()._random_topic_suffix()}'
+        self.password = f'secret_{self.username}'
+        self.consumer = None
+        self.topic = topic
+        self.produce_fut = None
+        self.fetch_fut = None
+
+
+class BasicAuthScaleTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(partition_count=3),
+        TopicSpec(partition_count=3),
+        TopicSpec(partition_count=3),
+        TopicSpec(partition_count=3),
+        TopicSpec(partition_count=3)
+    ]
+
+    def __init__(self, context):
+
+        security = SecurityConfig()
+        security.enable_sasl = True
+        security.endpoint_authn_method = 'sasl'
+        security.pp_authn_method = 'http_basic'
+
+        super(BasicAuthScaleTest, self).__init__(context, security=security)
+
+        self.users_list = []
+
+    @cluster(num_nodes=3)
+    @matrix(num_users=[10])
+    def test_many_users(self, num_users: int):
+        super_username, super_password, super_algorithm = self.redpanda.SUPERUSER_CREDENTIALS
+        rpk = RpkTool(self.redpanda)
+
+        for _ in range(num_users):
+            topic = random.choice(self.topics)
+            user = User(topic.name)
+            o = rpk.sasl_create_user(user.username, user.password,
+                                     super_algorithm)
+            print(o)
+            self.users_list.append(user)
+
+        for user in self.users_list:
+            o = rpk.sasl_allow_principal(f'User:{user.username}',
+                                         ['write', 'read', 'describe'],
+                                         'topic', user.topic, user.username,
+                                         user.password, super_algorithm)
+            print(o)
+            cc_res = self._create_named_consumer('g',
```

Comment:
This will fail because of `error_code: group_authorization_failed [30]` which implies that the user doesn't have perms to do group ops IIUC. I've tried adding `--group g` to the rpk command but that is apparently an unrecognized flag even though it's listed in the rpk help text. I'll ask someone about this on Monday